### PR TITLE
Get rid of evaluation_json_file in model 

### PR DESCRIPTION
**Pull request by OyvindTafjord**
This adds `answer_texts` to the metadata for SQuAD data so we can calculate official EM/F1 without the separate `evaluation_json_file` parameter. This has the added benefit of fixing `allennlp/run evaluate` on other validation sets (modulo vocabulary expansion which is a separate matter), as well as providing EM/F1 scores on training data as well. There was no noticeable slowdown when I checked a few epochs on a 10% sample.

Note that I left in a `params.pop('evaluation_json_file', None)` on the model loading for now, so we don't immediately invalidate the model files we have out there (for the demo etc).

Also tacked on here is the removal of `--` in the beaker script which is necessary for the latest version of beaker.

### CODE REVIEW

File: allennlp/data/dataset_readers/squad.py [hunk 1]
```diff
@@ -197,6 +200,8 @@ def text_to_instance(self,  # type: ignore
                 }
         if question_id:
             metadata['question_id'] = question_id
+        if answer_texts:
+            metadata['answer_texts'] = answer_texts
```
Comment (matt-gardner) [inline allennlp/data/dataset_readers/squad.py]: Can you add this to the docstring above, where we say what metadata is saved?

File: allennlp/data/dataset_readers/squad.py [hunk 2]
```diff
@@ -197,6 +200,8 @@ def text_to_instance(self,  # type: ignore
                 }
         if question_id:
```
Comment (matt-gardner) [inline allennlp/data/dataset_readers/squad.py]: Looks like we can remove the question id now.

File: allennlp/models/bidaf.py [hunk 3]
```diff
@@ -350,7 +329,7 @@ def from_params(cls, vocab: Vocabulary, params: Params) -> 'BidirectionalAttenti
         span_end_encoder = Seq2SeqEncoder.from_params(params.pop("span_end_encoder"))
         initializer = InitializerApplicator.from_params(params.pop("initializer", []))
         dropout = params.pop('dropout', 0.2)
-        evaluation_json_file = params.pop('evaluation_json_file', None)
+        params.pop('evaluation_json_file', None)  # Temporary, for backwards compatibility
```
Comment (matt-gardner) [inline allennlp/models/bidaf.py]: Maybe assign this to something and give a deprecation warning if it's not `None`?  And add a TODO to remove this once the saved models are updated?

File: allennlp/data/dataset_readers/squad.py (hunk 2 above)
Comment (OyvindTafjord) [reply allennlp/data/dataset_readers/squad.py]: I think it's nice to keep it there, e.g., I'm imagining using a version of this to dump individual scores per question into a file and then having easy access to the ID is handy.

File: allennlp/models/bidaf.py (hunk 3 above)
Comment (OyvindTafjord) [reply allennlp/models/bidaf.py]: OK, do we have an existing pattern for deprecation warnings yet?

File: allennlp/data/dataset_readers/squad.py (hunk 2 above)
Comment (matt-gardner) [reply allennlp/data/dataset_readers/squad.py]: Ok, fair enough.

File: allennlp/models/bidaf.py (hunk 3 above)
Comment (matt-gardner) [reply allennlp/models/bidaf.py]: No, we don't.

**Comment by OyvindTafjord** [comment]
Found one earlier deprecation warning in `allennlp/run` so followed that pattern, not that it's easy to see that one line in the stream of stuff coming up when loading, but at least it's there. Added the docstring stuff.

**Comment by matt-gardner** [comment]
Great.  And, sorry, I just merged a PR that will make you change this just a little bit.  I created a `MetadataField` that makes this metadata business a _little_ bit cleaner.

**Pull request merged**